lms/djangoapps/grades/new/subsection_grade.py

```diff
@@ -104,51 +93,66 @@ def load_from_data(self, model, course_structure, scores_client, submissions_sco
             section=self.display_name,
             module_id=self.location,
         )
+        self._log_event(log.warning, u"init_from_model", student)
 
-    def __unicode__(self):
+    @classmethod
+    def bulk_create_models(cls, student, subsection_grades, course_key):
         """
-        Provides a unicode representation of the scoring
-        data for this subsection. Used for logging.
+        Saves the subsection grade in a persisted model.
         """
-        return u"SubsectionGrade|total: {0}/{1}|graded: {2}/{3}|location: {4}|display name: {5}".format(
-            self.all_total.earned,
-            self.all_total.possible,
-            self.graded_total.earned,
-            self.graded_total.possible,
-            self.location,
-            self.display_name
+        return PersistentSubsectionGrade.bulk_create_grades(
+            [subsection_grade._persisted_model_params(student) for subsection_grade in subsection_grades],  # pylint: disable=protected-access
```

Comment:
I would suggest using a generator here, but I see that `bulk_create_grades()` iterates twice over the grades, so the list is necessary here